src/AppInstallerRepositoryCore/Rest/RestSource.cpp

```diff
@@ -0,0 +1,262 @@
+
+        protected:
+            std::shared_ptr<const RestSource> GetReferenceSource() const
+            {
+                std::shared_ptr<const RestSource> source = m_source.lock();
+                THROW_HR_IF(E_NOT_VALID_STATE, !source);
+                return source;
+            }
+
+        private:
+            std::weak_ptr<const RestSource> m_source;
+        };
+
+        // The IPackageVersion impl for RestSource.
+        struct PackageVersion : public SourceReference, public IPackageVersion
+        {
+            PackageVersion(
+                const std::shared_ptr<const RestSource>& source, IRestClient::PackageInfo packageInfo, VersionAndChannel versionInfo)
+                : SourceReference(source), m_packageInfo(packageInfo), m_versionInfo(versionInfo) {}
+
+            // Inherited via IPackageVersion
+            Utility::LocIndString GetProperty(PackageVersionProperty property) const override
+            {
+                switch (property)
+                {
+                case PackageVersionProperty::SourceIdentifier:
+                    return Utility::LocIndString{ GetReferenceSource()->GetIdentifier() };
+                case PackageVersionProperty::SourceName:
+                    return Utility::LocIndString{ GetReferenceSource()->GetDetails().Name };
+                case PackageVersionProperty::Id:
+                    return Utility::LocIndString{ m_packageInfo.packageIdentifier };
+                case PackageVersionProperty::Name:
+                    return Utility::LocIndString{ m_packageInfo.packageName };
+                case PackageVersionProperty::Version:
+                    return Utility::LocIndString{ m_versionInfo.GetVersion().ToString() };
+                case PackageVersionProperty::Channel:
+                    return Utility::LocIndString{ m_versionInfo.GetChannel().ToString() };
+                default:
+                    return Utility::LocIndString{};
+                }
+            }
+
+            // TODO
+            std::vector<Utility::LocIndString> GetMultiProperty(PackageVersionMultiProperty property) const override
+            {
+                UNREFERENCED_PARAMETER(property);
+                std::vector<Utility::LocIndString> result;
+                return result;
+            }
+
+            // TODO
+            Manifest::Manifest GetManifest() const override
+            {
+                AICLI_LOG(Repo, Info, << "Downloading manifest");
+                std::string manifest = GetReferenceSource()->GetRestClient().GetManifestByVersion(
+                    m_packageInfo.packageIdentifier, m_versionInfo.GetVersion().ToString()).value();
+
+                // TODO: Create Manifest object from result using JSON parser.
+                THROW_HR(ERROR_CALL_NOT_IMPLEMENTED);
+            }
+
+            std::shared_ptr<const ISource> GetSource() const override
+            {
+                return GetReferenceSource();
+            }
+
+            // TODO

```

Comment:
Metadata is currently only used for storing extra information about installed packages, so you likely don't need to do any more than this until we actually have some reason.